src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/DictionaryDefaultConverter.cs

```diff
@@ -102,11 +113,12 @@ protected static JsonConverter<TValue> GetValueConverter(ref WriteStack state)
                         }
 
                         state.Current.JsonPropertyNameAsString = reader.GetString();
```

Comment:
Do we still need to set `JsonPropertyNameAsString`?

Reply:
Since `DefaultDictionaryConverter.OnTryRead` is still being used by other dictionaries like `ImmutableDictionary` or `IDictionary`; right now it is still needed.

We can remove it once we spread the `TKey` changes to those converters as well.